src/coreclr/jit/rangecheck.cpp

```diff
@@ -1261,12 +1268,57 @@ bool RangeCheck::ComputeDoesOverflow(BasicBlock* block, GenTree* expr)
     {
         overflows = DoesPhiOverflow(block, expr);
     }
+    else if (expr->OperIs(GT_CAST) && expr->AsCast()->CastOp()->OperIs(GT_LCL_VAR))
+    {
+        // See if this is a cast on top of an argument - in this case we won't ever overflow
+        var_types      castType = expr->AsCast()->CastToType();
+        GenTreeLclVar* lcl      = expr->gtGetOp1()->AsLclVar();
+        if ((lcl->GetSsaNum() == SsaConfig::FIRST_SSA_NUM)) // make sure it's the first def of the arg
+        {
+            LclVarDsc* varDsc = m_pCompiler->lvaGetDesc(lcl->GetLclNum());
+            if (varDsc->lvIsParam && (varDsc->TypeGet() == castType) && varTypeIsSmall(castType))
+            {
+                overflows = false;
+            }
+        }
+    }
```

Comment:
When does a cast "overflow" by this definition of "overflow" in the general case?

Reply:
no idea, the whole function DoesOverflow which is able to "cancel" bound check elimination is questionable, but I don't want to remove as part of this PR